src/PowerShellEditorServices/Workspace/ScriptFile.cs

```diff
@@ -564,6 +579,43 @@ public BufferRange GetRangeBetweenOffsets(int startOffset, int endOffset)
 
         #region Private Methods
 
+        private static string GetPathAsClientPath(string path)
+        {
+            const string fileUriPrefix = "file:///";
+
+            if (path.StartsWith("untitled:", StringComparison.Ordinal))
+            {
+                return path;
+            }
+
+            if (path.StartsWith("file:///", StringComparison.Ordinal))
+            {
+                return path;
+            }
+
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return new Uri(path).AbsoluteUri;
+            }
+
+            // VSCode file URIs on Windows need the drive letter lowercase, and the colon
+            // URI encoded. System.Uri won't do that, so we manually create the URI.
+            var newUri = System.Web.HttpUtility.UrlPathEncode(path);
+            int colonIndex = path.IndexOf(":");
```

Comment:
I've already written logic for this somewhere in PSES I think, so ideally we should use that as a static method

Reply:
Also, I opened an issue on VSCode about it and they tried changing it (to their credit) and it didn't work out. The URI spec says to use `:` for drive letters, but alas...

Reply:
Here: https://github.com/PowerShell/PowerShellEditorServices/blob/1a86d4b159c17f382aa148942598e140d7c0070d/src/PowerShellEditorServices/Workspace/Workspace.cs#L595-L626

Reply:
Oh we need the opposite method!

Reply:
We should stick them in the same place

Reply:
Done.